index.md

```diff
@@ -19,14 +19,31 @@ GET /jsonapi/node/article/{id}
 GET /jsonapi/node/article/{id}
 Accept-Language: fr
 
-200 OK
-
 // If translation exists:
+200 OK
 Content-Language: fr
 Content-Location: /jsonapi/node/article/{id}?lang=fr
 
 // If translation does not exist:
+// option a)
+// provide a default without redirecting
+200 OK
 Content-Language: en
+
+// option b)
+// do not provide a default or redirect, simply provide alternatives
+406 Not acceptable
```

Comment:
I think I prefer this option: it seems it would force clients to acknowledge that the previous request did not yield a valid result and that a further (explicit) action is necessary. Would it be possible to still provide the URL of the default translation through the `Content-Location` header, even with a `406` status?

Reply:
I think it would be pretty unconventional to provide a Location with a 406.

Honestly, my preference is for option C. 4xx errors are client errors, but it's not really a client error to ask for a language that the server doesn't have and a 300 will still force the client to make a better request. Clients can choose to follow the redirection or not, it's not forced.